### ISSUE TITLE
Does `hough_line`'s `min_angle` parameter function correctly?

### ISSUE THREAD

**Issue by stefanv** [scikit-image/scikit-image#2765]
From https://stackoverflow.com/questions/45782927/what-is-the-meaning-of-min-distance-and-min-angle-in-hough-line-peaks

> Can someone explain min_distance and min_angle optional parameters, please ?
>
> http://scikit-image.org/docs/dev/api/skimage.transform.html#skimage.transform.hough_line_peaks
>
> For `min_angle=n`, I thought it would check if the next angle's line was minimum superior to n element in my theta array for being accepted.

```
import numpy as np
from skimage.transform import hough_line,hough_line_peaks

iden = np.identity(200)
hspace, angles, dists = hough_line(iden,theta=np.linspace(-np.pi/2,np.pi/2,1800)) # 0.1 degree resolution
hspace, angles, dists = hough_line_peaks(hspace, angles, dists,min_distance=0,min_angle=20) # 2 degree minimum before accepting as new line?  
print(hspace, angles*180/np.pi, dists)
```

```
output : [200 126 124] [-44.9749861 -45.27515286 -44.67481934] [ 0.50088496 -0.50088496 1.50265487]
```

> The angle array shows that i'm getting wrong with this. The parameter accepts only integer, i'm not sure of what it could be ...

**Comment by guihh** [comment]
hey, i'm the author of the stackoverflow's post. i found that using `theta = np.linspace(-np.pi/2,np.pi/2,1801)`, i get value's angle with true 0.1 degree resolution. Is this normal ? ( you are using 180 as default value for theta but **i'm not getting whole value** when converted to degree, but i get whole value with 181 ...)

```python
import numpy as np
from skimage.transform import hough_line,hough_line_peaks

iden = np.identity(200)
hspace, angles, dists = hough_line(iden,theta=np.linspace(-np.pi/2,np.pi/2,1801)) # 0.1 degree resolution
hspace, angles, dists = hough_line_peaks(hspace, angles, dists,min_distance=0,min_angle=20) # 2 degree minimum before accepting as new line?  
print(hspace, angles*180/np.pi, dists)
```

`output : [200, 132, 132] [-45. , -44.7, -45.3] [ 0.50088496,  1.50265487, -0.50088496]`

I saw min_angle refers to min_xdistance in _prominent_peak() in [scikit.feature.peak](https://github.com/scikit-image/scikit-image/blob/master/skimage/feature/peak.py) which is passed to scipy.ndimage.filters.maximum_filter1d(). 

Looking to the code, this value is used as **half of the filter size** for, first finding maximum and then suppressing the potential neighbours of the filtered peaks, isn't it ?

thanks in advance,

Guillaume H.

**Comment by stefanv** [comment]
As user Eskapp explained on S/O, this may be a documentation issue, rather than an actual error in the implementation:

https://stackoverflow.com/a/46198201/214686

**Comment by Eskapp** [comment]
Hello,
I am the person who wrote the S/O answer. I can help clarifying the documentation if someone explains to me how I can do so.

**Comment by guihh** [comment]
Hey,

Thank you for your well explained answer on S-O ! But i'm still confused with the output of houghline_peaks() ...

```python
import numpy as np
from skimage.transform import hough_line,hough_line_peaks

iden = np.identity(200)
hspace, angles, dists = hough_line(iden,theta=np.linspace(-np.pi/2,np.pi/2,1801)) # 0.1 degree resolution
hspace, angles, dists = hough_line_peaks(hspace, angles, dists,min_distance=0,min_angle=50) # 0.5 degree thresholding.
print(hspace, angles*180/np.pi, dists)
```
output [200 132 132] [-45.  -44.7 -45.3] [ 0.50088496  1.50265487 -0.50088496]

This should output only [200] [-45.] [ 0.50088496], according to [bin - min_dist: bin + min_dist, bin - min_angle:bin + min_angle] ... Am i wrong ?

**Comment by Eskapp** [comment]
No, all the returned distances are different. `min_distance=0` will only impacts peaks that would lie in a bin corresponding to the exact same distance parameter than a previously found peak. I don't know what the default distance resolution is **(the documentation should precise this too by the way)**.

If you set `min_distance` to a value that is big enough, keeping the same `min_angle`, the third peak should be discarded.

**Comment by guihh** [comment]
Why using min_angle=50 (in my exemple 0.5degree) won't discard the 2 others peaks ? I saw that _prominent_peak() used in hough_line_peaks() was using filtermaximum1d() from scipy so it should discard it regardless of the min_distance parameter ? 

Thank for your answer, i've also asked the question on S-O.

**Comment by stefanv** [comment]
@Eskapp That you for your willingness to help out here; much appreciated.  I will very happily help you with the Git process.  Here's our guide for new contributors:

http://scikit-image.org/docs/dev/contribute.html

If there are any details missing from that document, just let me know and I'll update it.  Thanks!
